core/lib/spree/migration_helpers.rb

```diff
@@ -0,0 +1,63 @@
+module Spree
+  module MigrationHelpers
+    class SafeIndexChange
+      attr_reader :table, :columns, :options
+
+      def initialize(table, columns, options = {})
+        @table = table
+        @columns = columns
+        @options = options
+      end
+
+      def remove_index
+        remove_index(table, columns) if index_present?
+      end
+
+      def add_index
+        if columns.is_a?(Array)
+          add_multi_column_index
+        else
+          add_single_column_index
+        end
+      end
+
+      private
+
+      def index_present?
+        index_exists?(table, columns, options)
+      end
+
+      def index_absent?
+        !index_present?
+      end
+
+      def add_multi_column_index
+        if all_columns_exist? && index_absent?
+          add_index(table, columns, options)
+        end
+      end
+
+      def all_columns_exist?
+        columns.all? { |column| column_exists?(column) }
```

Comment:
Note we discard the `options` parameter when checking for the existence of multiple columns. I believe this is a reasonable compromise. What do you think?